src/physics/grid/PhysicsGridCalculator.i.hh

```diff
@@ -28,50 +28,48 @@ PhysicsGridCalculator::operator()(const ParticleTrackView& particle) const
 //---------------------------------------------------------------------------//
 /*!
  * Calculate the cross section.
- *
- * Assumes that the energy grid has the same units as particle.energy.
- *
- * XXX also this breaks if prime_energy != 1, since the
- * value of "xs" is actually xs*E above E' but the stored value at the lower
- * grid point is just xs.
- *
- * To fix that, we can change 'prime_energy' to 'prime_energy_index' since it
- * should always be on a grid point. If `bin == prime_energy_index`, then scale
- * the lower xs value by E. If bin >= prime_energy_index, scale the result by
- * 1/E.
  */
 CELER_FUNCTION real_type PhysicsGridCalculator::operator()(Energy energy) const
 {
     UniformGrid loge_grid(data_.log_energy);
     real_type   loge = std::log(energy.value());
 
     // Snap out-of-bounds values to closest grid points
-    size_type bin;
+    size_type lower_idx;
     real_type result;
     if (loge <= loge_grid.front())
     {
-        bin    = 0;
+        lower_idx = 0;
         result = data_.value.front();
     }
     else if (loge >= loge_grid.back())
     {
-        bin    = data_.value.size();
+        lower_idx = data_.value.size();
```

Comment:
I still think this should be `lower_idx = data_.value.size() - 1`.. or am I missing something?

Reply:
Uh I think you're right. Off-by-one error in my head.